lib/config/flat-config-array.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * @fileoverview Flat Config Array
+ * @author Nicholas C. Zakas
+ */
+
+"use strict";
+
+//-----------------------------------------------------------------------------
+// Requirements
+//-----------------------------------------------------------------------------
+
+const { ConfigArray, ConfigArraySymbol } = require("@humanwhocodes/config-array");
```

Comment:
Will `ConfigArray` support negated patterns (prefixed with `!`) in `files` and/or `ignores`?

Reply:
It already does

Reply:
What's the expected behavior in this case:

```js
{
    files: ["*.js", "!foo.js"]
}
```

By the RFC, it might be expected that the config applies to all `*.js` files except `foo.js`:

> These patterns can be negated by prefixing them with `!`, which effectively mimics the behavior of `excludedFiles` in `.eslintrc`.

However, `ConfigArray` checks if the given file path matches at least one of the patterns in the `files` array, so the config will apply to any file path (`a.js`, `foo.js`, `a.ts`, `package.json`...).

Reply:
This behavior is pretty well-defined in the RFC: `files` are treated as OR statements. If you want to do an AND, then you need to group them in an array:

```
{
    files: [["*.js", "!foo.js"]]
}
```